src/System.Collections/tests/Performance/Perf.Dictionary.cs

```diff
@@ -193,5 +193,356 @@ public void ContainsKey(int size)
+            long?[] items;
+
+            items = new long?[size * 10];
+
+            for (long i = 0; i < size * 10; ++i)
+            {
+                items[i] = i;
+                s_collection1.Add(items[i], items[i]);
+            }
+
+
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (long i = 1; i < size; ++i)
+                        s_collection1.Remove(items[i]);
+        }
+
+        [Benchmark]
+        public static void Indexer_get_ValueType()
+        {
+            int size = 1024;
+
+            s_collection = new Dictionary<int?, int?>();
+            for (int i = 0; i < size; ++i)
+            {
+                s_collection.Add(i, i);
+            }
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int j = 0; j < size; ++j)
+                    {
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        public static void Enumeration_ValueType()
+        {
+            int size = 1024;
+            s_collection = new Dictionary<int?, int?>();
+
+            for (int i = 0; i < size; ++i)
+            {
+                s_collection.Add(i, i);
+            }
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    foreach (KeyValuePair<int?, int?> tempItem in s_collection)
+                    {
+                        s_key = tempItem.Key;
+                        s_value = tempItem.Value;
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        public static void Add_ValueType()
+        {
+            int size = 2048;
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                s_collection = new Dictionary<int?, int?>();
+
+                using (iteration.StartMeasurement())
+                {
+                    for (int j = 0; j < size; ++j)
+                    {
+                        s_collection.Add(j, j);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        public static void Dictionary_ContainsValue_Int_True()
```

Comment:
Yes, this is a duplicate.  I've gone ahead and removed the existing ContainsKey test, but modified the new ContainsKey and ContainsValue tests to take variable data and run lengths like the existing ContainsKey test.